### ISSUE TITLE
Undo Button for Un-Merging Molecule Entries?

### ISSUE THREAD

**Issue by thoretz** [duderstadt-lab/mars-fx#79]
I was merging two molecule entries by accident and was unable to undo the action. This forced me to start over with my archive and redo some of the steps. 

Would there be a way to make this process reversible? Another option would be to keep the old molecule entries and tag them for later analysis. Like 'partial trace' for the incomplete trace and 'merged' for the merged trace. I am manually marking traces that are merged with the 'merged' tag so I can quickly asses which ones were merged. 

Thanks a lot!

**Comment by karlduderstadt** [comment]
Yes, I like the idea of keeping the unmerged molecule records. I think this is the easiest way to allow an undo. The merge command requires that you tag some records with a common tag to be merged. What tag do you use for that?

I think it makes sense to add partial trace for molecules that were merged and could be deleted. So we can implement that.

However, I am puzzled what is the best way to deal with merged molecules. We could certainly add the merged tag, but is that also the tag you use when merging or how do you do it?

**Comment by thoretz** [comment]
For merging, I tag the molecules which I want to merge with 'merge' (2 or more sometimes). Then I get the newly merged trace and I change the tag to 'merged'. But this is very flawed because forgetting to change the tag will lead to merged molecule traces which a merged into another set of traces. In the past, I used the first five characters of the UID which made it unique but also more tedious. 

Maybe after merging the tag could automatically change to 'merged'. Would that be an option?
Thank you!

**Comment by karlduderstadt** [comment]
Yes, I think it is a good idea to have the tag change automatically. I think maybe I would add options to the dialog for the final tag. Now I have a clear idea what needs to be done.
